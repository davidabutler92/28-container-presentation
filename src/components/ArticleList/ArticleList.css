.container {
  border: solid 4px red;
}