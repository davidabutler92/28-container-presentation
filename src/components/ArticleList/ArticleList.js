import React from 'react';
import ArticleItem from '../ArticleItem/ArticleItem';
import PropTypes from 'prop-types';

function ArticleList({ articles }) {
  const articlesList = articles.map((article) => {
    return (
      <li key={article.id + Date.now()}>
        <ArticleItem
          title={article.title}
          author={article.author}
          description={article.description}
        />
      </li>
    );
  });

  return <ul>{articlesList}</ul>;
}

ArticleList.propTypes = {
  articles: PropTypes.array,
  source: PropTypes.object,
};

export default ArticleList;
