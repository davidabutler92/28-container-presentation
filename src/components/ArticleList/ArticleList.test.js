import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticelList component', () => {
  afterEach(() => cleanup());
  it('renders a list of articles', () => {
    const data = [
      { 
        id: 1,
        title: '',
        author: '',
        description: '',
      },
      { 
        id: 2,
        title: '',
        author: '',
        description: '',
      },
      { 
        id: 3,
        title: '',
        author: '',
        description: '',
      },
    ];

    const { asFragment } = render(<ArticleList articles={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
