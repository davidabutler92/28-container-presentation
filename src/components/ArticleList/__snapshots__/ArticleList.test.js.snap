// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticelList component renders a list of articles 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <h1 />
      <p />
      <p />
    </li>
    <li>
      <h1 />
      <p />
      <p />
    </li>
    <li>
      <h1 />
      <p />
      <p />
    </li>
  </ul>
</DocumentFragment>
`;
